### PR TITLE
Update minimum required Ruby version

### PR DESCRIPTION
**Pull request by MuriloDalRi**
We're adding support for Ruby 3.3 and dropping support for 3.0. The new minimum required version will be 3.1.4.

For more info: https://docs.publishing.service.gov.uk/manual/publishing-a-ruby-gem.html#ruby-version-compatibility

### CODE REVIEW

File: govuk_ab_testing.gemspec [hunk 1]
```diff
@@ -12,7 +12,7 @@ Gem::Specification.new do |spec|
   spec.description   = "Gem to help with A/B testing on the GOV.UK platform"
   spec.homepage      = "https://github.com/alphagov/govuk_ab_testing"
   spec.license       = "MIT"
-  spec.required_ruby_version = ">= 3.0"
+  spec.required_ruby_version = ">= 3.1.4"
```
Comment (kevindew) [inline govuk_ab_testing.gemspec:15]:
```suggestion
  spec.required_ruby_version = ">= 3.1"
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+## Unreleased
+
+* Drop support for Ruby 3.0. The minimum required Ruby version is now 3.1.4.
+* Add support for Ruby 3.3.
```
Comment (kevindew) [inline CHANGELOG.md:4]:
```suggestion
```
We've not actually changed the code for this, so we don't need an entry

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
+## Unreleased
+
+* Drop support for Ruby 3.0. The minimum required Ruby version is now 3.1.4.
```
Comment (kevindew) [inline CHANGELOG.md:3]:
```suggestion
* Drop support for Ruby 3.0. The minimum required Ruby version is now 3.1.
```

It should support all version of 3.1, not necessarily the most recent patch.

**Review by kevindew — COMMENTED** [review]
Thanks @MuriloDalRi, super helpful. I've put a few suggestions for things to fix here which you can apply to the next batch one.

**Review by davidgisbey — APPROVED** [review]
lgtm!

**Pull request merged**